objects/src/accounts/storage/mod.rs

```diff
@@ -158,7 +158,7 @@ impl AccountStorage {
     /// Returns a new instance of account storage initialized with the provided items.
     pub fn new(
         items: Vec<SlotItem>,
-        maps: Vec<StorageMap>,
+        maps: BTreeMap<u8, StorageMap>,
```

Comment:
If we are passing `maps` as `BTreeMap<u8, StorageMap>` we should also validate that the indexes are correct (i.e., match the layout).

I think a better option may be to keep `maps` as `Vec<StorageMap>` and convert to the `BTreeMap` internally in the constructor based on the layout info.

Reply:
I validate the index now. 

If we keep using a `Vec<StorageMap>`, I can't identify the correct `StorageMap` to which I want to do the update. The problem is, that I cannot use the root of the map as identifier, because the corresponding `StorageSlot` is also updated by `apply_delta`. So it depends on the order which delta is being applied first. A more secure way is to identify the `StorageMap` by index.

Reply:
I think we are always identifying storage maps by index. But I'm fine with keeping `BTreeMap` in the constructor signature (and we'll probably refactor this anyway as a part of the work that @phklive is doing).